dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -133,27 +133,52 @@ internal KernelFunction(string name, string description, IReadOnlyList<KernelPar
             cancellationToken.ThrowIfCancellationRequested();
 
             // Invoke pre-invocation event handler. If it requests cancellation, throw.
-            if (kernel.OnFunctionInvoking(this, arguments)?.Cancel is true)
+            var invokingEventArgs = kernel.OnFunctionInvoking(this, arguments);
+
+            // Invoke pre-invocation filter. If it requests cancellation, throw.
+            var invokingContext = kernel.OnFunctionInvokingFilter(this, arguments);
+
+            if (invokingEventArgs?.Cancel is true)
             {
                 throw new OperationCanceledException($"A {nameof(Kernel)}.{nameof(Kernel.FunctionInvoking)} event handler requested cancellation before function invocation.");
             }
 
+            if (invokingContext?.Cancel is true)
```

Comment:
I find it strange to have both systems and they don't interact, eg an event handler can cancel cancelation by setting the property back to false, but a filter can't.

Reply:
So, if we use only event handlers or only filters, it is possible to cancel cancellation. But if we cancel cancellation in event handler, but we want to proceed with cancellation in filter - it's not going to work, and I'm not sure if it's a goal to be honest. I think the primary usage pattern should be to use only event handlers or only filters, based on personal preference, but not combine them together.

Reply:
Resolving this as events will be removed in the future, so no need to make both approaches interact with each other.